### ISSUE TITLE
importing a mongodbatlas_project_ip_whitelist resource does not save project_id to state

### ISSUE THREAD

**Issue by dlethin** [mongodb/terraform-provider-mongodbatlas#21]
There seems to be a bug with importing `mongodbatlas_project_ip_whitelist` resources.  The `project_ip` attribute is not saved in state, so the next time you run terraform, it wants to destroy and create the resource because project_id is not set.

Here is some example code showing this:

```
provider "mongodbatlas" {}

resource "mongodbatlas_project" "project" {
    name   = "${var.atlas_project_name}"
    org_id = "${var.atlas_organization}"
}

resource "mongodbatlas_project_ip_whitelist" "whitelist" {
  project_id = "${mongodbatlas_project.project.id}"

  whitelist {
    ip_address = "1.2.3.4"
  }
}
```

I have no problem importing `mongodbatlas_project.project` &
`mongodbatlas_project_ip_whitelist.whitelist`

But then when I run `terraform plan`, terraform indicates that it wants to destroy and create the resource because project_id is not set:

```
$ terraform plan --target mongodbatlas_project_ip_whitelist.whitelist
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

mongodbatlas_project.project: Refreshing state... [id=<MY_PROJECT_ID>]
mongodbatlas_project_ip_whitelist.whitelist: Refreshing state... [id=<XXXXXXX>==]

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # mongodbatlas_project_ip_whitelist.whitelist must be replaced
-/+ resource "mongodbatlas_project_ip_whitelist" "whitelist" {
      ~ id         = "<XXXXXXX>" -> (known after apply)
      + project_id = "<MY_PROJECT_ID>" # forces replacement

      + whitelist { # forces replacement
          + cidr_block = (known after apply)
          + comment    = (known after apply)
          + ip_address = "1.2.3.4"
        }
    }

Plan: 1 to add, 0 to change, 1 to destroy.```

This is using version 0.1.1 of the plugin.

As a side note, I also have a `mongodbatlas_cluster` I can import and that works fine. The project_id is saved in state there.

Thanks.

**Comment by dlethin** [comment]
I started trying to try to fix this on my own and submit a PR, but I'm running in to some trouble.  Following the instructions in the README, I get the following error on the `make build` step:

```
go: git.apache.org/thrift.git@v0.0.0-20180902110319-2566ecd5d999: git fetch -f origin refs/heads/*:refs/heads/* refs/tags/*:refs/tags/* in <HOME>/bld/go/pkg/mod/cache/vcs/83dba939f95a790e497d565fc4418400145a1a514f955fa052f662d56e920c3e: exit status 128:
	fatal: unable to access 'https://git.apache.org/thrift.git/': Failed to connect to git.apache.org port 443: Operation timed out
```

not sure how deep in the dependency chain this is coming from and not a seasoned go-lang dev to immediate dive further into this.

In the meantime, this is the diff I think needs to change to resolve the issue I reported:

```
diff --git a/mongodbatlas/resource_mongodbatlas_project_ip_whitelist.go b/mongodbatlas/resource_mongodbatlas_project_ip_whitelist.go
index 56c91e7..d7d0dab 100644
--- a/mongodbatlas/resource_mongodbatlas_project_ip_whitelist.go
+++ b/mongodbatlas/resource_mongodbatlas_project_ip_whitelist.go
@@ -181,6 +181,11 @@ func resourceMongoDBAtlasProjectIPWhitelistImportState(d *schema.ResourceData, m
        }))

        if err := d.Set("whitelist", flattenProjectIPWhitelist(whitelist)); err != nil {
+               log.Printf("[WARN] Error setting whitelist for (%s): %s", d.Id(), err)
+               return []*schema.ResourceData{d}, err
+       }
+
+       if err := d.Set("project_id", d.Id()); err != nil {
                log.Printf("[WARN] Error setting project_id for (%s): %s", d.Id(), err)
                return []*schema.ResourceData{d}, err
        }
```
This is essentially mimicking what was done [here](https://github.com/terraform-providers/terraform-provider-mongodbatlas/blob/master/mongodbatlas/resource_mongodbatlas_cluster.go#L583) for mongodbatlas_cluster 
cheers.

**Comment by PacoDw** [comment]
Hello, @dlethin thank you so much for your review 👍 , let us check to resolve the issue.

**Comment by PacoDw** [comment]
Yeah! your changes are right, now you can test it on pr #23, thank you!

If you have any comments or another issue just let us know.

**Comment by dlethin** [comment]
Thanks for the quick response.  

I'm not sure how to test it though, as I was running into problems building the plugin locally as mentioned above.  Maybe I just don't have my go-lang environment set up correctly..

**Comment by PacoDw** [comment]
Don't worry the changes are now in master, you could clone the repository and test it directly once you have the repository downloaded just make the following command inside of the folder:

```$ go build```  
This will generate a binary program that it is the plugin that you could use with terraform directly, move this binary in your terraform project folder.

**Comment by dlethin** [comment]
I built it locally and confirmed it works for my project.  

I did discover another problem where adding an additional whitelist ip entry then wants to destroy and rebuild the resource rather than incrementally changing which is not ideal as there will be a moment of time where existing atlas clients might get shut out until their whitelist entries are added again, but I can track a separate issue for that.

Thanks for sharing this plugin. Cheers.
